framework/src/media/audio/audio_manager.c

```diff
@@ -563,15 +576,20 @@ int start_audio_stream_in(void *data, uint32_t frames)
 			ret = pcm_prepare(pcm);
 			if (ret != OK) {
 				meddbg("Fail to pcm_prepare()\n");
-				return AUDIO_MANAGER_XRUN_STATE;
+				ret = AUDIO_MANAGER_XRUN_STATE;
+				break;
 			}
 		} else if (ret == -EINVAL) {
-			return AUDIO_MANAGER_INVALID_PARAM;
+			ret = AUDIO_MANAGER_INVALID_PARAM;
+			break;
 		} else {
-			return ret;
+			break;
 		}
 	} while ((ret == OK) && (retry--));
 
+error_out:
```

Comment:
instead of error_out, please rename error_with_mutex error_out can be used in future.

Reply:
Ok, I'll change.